### PR TITLE
384 prop names prop types

### LINKED ISSUE #384: add common test to make sure `childComponent`s' `propName`s are included in `propTypes`

### PR DESCRIPTION
**Pull request by sodiumjoe**
fixes #384 

## PR Checklist

- Manually tested across supported browsers
  - [x] Chrome
  - ~~[ ] Firefox~~
  - ~~[ ] Safari~~
  - ~~[ ] IE11 (Win 7)~~
  - ~~[ ] Edge (Win 10)~~
- [x] Unit tests written (`common` at minimum)
- [x] PR has one of the `semver-` labels
- [ ] Two core team engineer approvals
- ~~[ ] One core team UX approval~~

### CODE REVIEW

**Comment by sodiumjoe** [comment]
I made a lot of judgement calls on which child components should have `propName`s. The ones i took out were not documented, and didn't have any examples associated, so I'm calling this a `semver-none`.

File: src/components/ButtonGroup/ButtonGroup.jsx [hunk 1]
```diff
@@ -27,7 +27,6 @@ const ButtonGroup = createClass({
 	components: {
 		Button: createClass({
 			displayName: 'ButtonGroup.Button',
-			propName: 'Button',
```
Comment (jondlm) [inline src/components/ButtonGroup/ButtonGroup.jsx]: What is the implication of removing a `propName`? Does that mean the child component shouldn't be pass in as a prop, but only as a child?

Comment (sodiumjoe) [inline src/components/ButtonGroup/ButtonGroup.jsx]: correct

File: src/components/ToolTip/ToolTip.jsx [hunk 2]
```diff
@@ -108,15 +108,12 @@ const ToolTip = createClass({
 	components: {
 		Target: createClass({
 			displayName: 'ToolTip.Target',
-			propName: 'Target',
 		}),
 		Title: createClass({
 			displayName: 'ToolTip.Title',
-			propName: 'Title',
```
Comment (jondlm) [inline src/components/ToolTip/ToolTip.jsx]: Doesn't this make sense as a prop?

File: src/components/ToolTip/ToolTip.jsx [hunk 3]
```diff
@@ -108,15 +108,12 @@ const ToolTip = createClass({
...
 		}),
 		Body: createClass({
 			displayName: 'ToolTip.Body',
-			propName: 'Body',
```
Comment (jondlm) [inline src/components/ToolTip/ToolTip.jsx]: Same here, what if you just want text?

Comment (sodiumjoe) [inline src/components/ToolTip/ToolTip.jsx]: The weird thing with these is that these wouldn't be props of `ToolTip`, they'd be props of a `ToolTip.FlyOut`, which would actually be the child component if this were factored correctly, but that depends on a refactor of `ContextMenu`.

Comment (jondlm) [inline src/components/ToolTip/ToolTip.jsx]: Gotcha, sounds good

**Pull request merged**